### PR TITLE
[AMSDK-8378] - Places Stop API with boolean to clear places data

### PR DESCRIPTION
**Pull request by PravinPK**
## Description

Modified the stop API to accept a boolean param. If YES is passed, all Places data will be removed from the device.

## Motivation and Context

Location data becomes stale quickly if it is no longer being updated. Stale data can lead to false positives from the Rules Engine and improper triggering of location-related experiences. Providing an option to clear data at the time location monitoring is stopped will prevent false positives due to stale location data.

## How Has This Been Tested?

Unit tests. Manual tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/places-monitor-android/src/phone/java/com/adobe/marketing/mobile/PlacesMonitorConstants.java [hunk 1]
```diff
@@ -23,7 +23,6 @@
 	static final String EXTENSION_NAME = "com.adobe.placesMonitor";
 
 	static final String EXTENSION_VERSION = "1.0.2";
```
Comment (PravinPK) [inline code/places-monitor-android/src/phone/java/com/adobe/marketing/mobile/PlacesMonitorConstants.java:25]: Did some cleanup on the constants file... Removed all the unused constants

File: code/places-monitor-android/src/phone/java/com/adobe/marketing/mobile/PlacesMonitorInternal.java [hunk 2]
```diff
@@ -262,7 +262,15 @@ private void processMonitorRequestEvent(final Event event) {
...
-			stopMonitoring();
+
+			EventData data = event.getData();
+			if(data != null && !data.isEmpty()){
+				boolean shouldClear = data.optBoolean(PlacesMonitorConstants.EventDataKeys.EVENT_DATA_CLEAR, false);
+				stopMonitoring(shouldClear);
+				return;
+			}
+
+			stopMonitoring(false);
```
Comment (PravinPK) [inline code/places-monitor-android/src/phone/java/com/adobe/marketing/mobile/PlacesMonitorInternal.java]: if an stop event is dispatched with empty eventData then the places states wont be cleared

File: code/places-monitor-android/src/phone/java/com/adobe/marketing/mobile/PlacesGeofenceManager.java [hunk 3]
```diff
@@ -155,7 +155,7 @@ void startMonitoringFences(List<PlacesPOI> nearByPOIs) {
 	/**
 	 * Stops monitoring for entry and exit event on nearby places of interest.
 	 */
-	void stopMonitoringFences() {
+	void stopMonitoringFences(final boolean clearData) {
```
Comment (sbenedicadb) [inline code/places-monitor-android/src/phone/java/com/adobe/marketing/mobile/PlacesGeofenceManager.java:163]: please update the javadoc to account for the new parameter and behavior

File: code/places-monitor-android/src/phone/java/com/adobe/marketing/mobile/PlacesMonitorInternal.java [hunk 4]
```diff
@@ -262,7 +262,15 @@ private void processMonitorRequestEvent(final Event event) {
 		if (PlacesMonitorConstants.EVENTNAME_START.equals(eventName)) {
 			startMonitoring();
 		} else if (PlacesMonitorConstants.EVENTNAME_STOP.equals(eventName)) {
-			stopMonitoring();
+
+			EventData data = event.getData();
+			if(data != null && !data.isEmpty()){
+				boolean shouldClear = data.optBoolean(PlacesMonitorConstants.EventDataKeys.EVENT_DATA_CLEAR, false);
```
Comment (sbenedicadb) [inline code/places-monitor-android/src/phone/java/com/adobe/marketing/mobile/PlacesMonitorInternal.java]: let's refactor this code so we don't have to use a `return` and we only have code to call `stopMonitoring` once

**Review by sbenedicadb — CHANGES_REQUESTED** [review]
looks good, a couple minor changes needed

File: code/places-monitor-android/src/phone/java/com/adobe/marketing/mobile/PlacesMonitorInternal.java [hunk 5]
```diff
@@ -262,7 +262,14 @@ private void processMonitorRequestEvent(final Event event) {
...
 		} else if (PlacesMonitorConstants.EVENTNAME_STOP.equals(eventName)) {
-			stopMonitoring();
+
+			boolean shouldClear = false;
+			EventData data = event.getData();
+			if(data != null && !data.isEmpty()){
+				shouldClear = data.optBoolean(PlacesMonitorConstants.EventDataKeys.EVENT_DATA_CLEAR, false);
+			}
+
+			stopMonitoring(shouldClear);
```
Comment (sbenedicadb) [inline code/places-monitor-android/src/phone/java/com/adobe/marketing/mobile/PlacesMonitorInternal.java:272]: 👍

**Review by sbenedicadb — APPROVED** [review]

**Pull request merged**